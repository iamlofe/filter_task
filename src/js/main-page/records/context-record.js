import { Record } from 'immutable';

const CurrentContext = new Record({
    searchTitle: null,
    searchType: null,
    demesions: null,
    id: null,

});

export default class CurrentContextRecord extends CurrentContext {
    static parse(info) {
        return new CurrentContextRecord({
            title: info.title,
            id: info.id,
            searchTitle: info.searchTitle,
            searchType: info.searchType
        });
    }
}
