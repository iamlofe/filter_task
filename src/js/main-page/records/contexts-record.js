import { Record, List } from 'immutable';

import RecordDemisions from './dimensions-record';

const Context = new Record({
    id: null,
    title: null,
    listsDimensions: new List(),
    isAccess: true,
    isActive: false,
});

export default class RecordContext extends Context {
    static parse(context) {
        return new RecordContext({
            id: context.id,
            title: context.title,
            listsDimensions:
                new List(context.listsDimensions.map(dimension => RecordDemisions.parse(dimension, context.id))),
        });
    }
}
