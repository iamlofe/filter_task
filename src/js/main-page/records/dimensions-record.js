import {
    Record,
    List,
} from 'immutable';

import RecordResults from './results-record';

const Demisions = new Record({
    id: null,
    title: null,
    listsResults: new List(),
    isActive: false,
    isAccess: false,
    demisionsSave: []
})

export default class RecordDemisions extends Demisions {
    static parse(demisions, contextId) {
        return new RecordDemisions({
            id: demisions.id,
            title: demisions.title,
            contextId: contextId,
            listsResults: new List(demisions.listResults.map(r => RecordResults.parse(r, contextId, demisions.id))),
        })
    }
}