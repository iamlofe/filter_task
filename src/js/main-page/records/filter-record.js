import { Record } from 'immutable';

export const CurrentContext = new Record({
    searchTitle: null,
    searchType: null,
    isSaving: null,
    isRestoring: null,
    contextIds: null
});
