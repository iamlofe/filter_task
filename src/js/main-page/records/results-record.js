import {
    Record,
    List,
} from 'immutable';

import idGeneration from '../tools/generation-id-tool';

const Results = new Record({
    id: null,
    title: null,
    demisionId: null,
    contextId: null,
    isActive: false,
    isAccess: false,
    resultsSave: []
})


export default class RecordResults extends Results {
    static parse(result, ctxId, dimId) {
        return new RecordResults({
            id: result.id,
            title: result.title,
            demisionId: dimId,
            contextId: ctxId
        })
    }
}