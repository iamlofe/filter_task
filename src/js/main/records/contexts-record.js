import { Record, List } from 'immutable';

import RecordDemisions from './dimensions-record';

export const Context = new Record({
    id: null,
    contextId: null,
    title: null,
    listsDimensions: new List()
});

export default class ContextRecord extends Context {
    static parse(context) {
        return new ContextRecord({
            id: context.id,
            contextId: context.id,
            title: context.title,
            listsDimensions: new List(context.listsDimensions.map(dimension => RecordDemisions.parse(dimension, context.id)))
        });
    }
}
