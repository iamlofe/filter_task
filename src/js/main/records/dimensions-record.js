import { Record, List } from 'immutable';

import RecordResults from './results-record';

const Demisions = new Record({
    id: null,
    title: null,
    contextId: null,
    listsResults: new List(),
    demisionId: null
});

export default class DemisionsRecord extends Demisions {
    static parse(demisions, contextId) {
        return new DemisionsRecord({
            demisionId: demisions.id,
            title: demisions.title,
            contextId,
            listsResults: new List(demisions.listResults.map(r =>
                RecordResults.parse(r, contextId, demisions.id)))
        });
    }
}
