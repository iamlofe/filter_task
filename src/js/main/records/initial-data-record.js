import { Record, List } from 'immutable';

export default new Record({
    initialDataFilter: new List(),
    isStatusLoadData: true
});
