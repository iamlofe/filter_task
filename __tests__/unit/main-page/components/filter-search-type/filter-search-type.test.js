import React from 'react';
import { shallow } from 'enzyme';
import { List } from 'immutable';

import FilterSearchType from 'main-page/components/filter-search-type/filter-search-type';
import { CurrentContext } from 'main-page/records/filter-record';
import { getMockListItems } from '../../mock-store/mock-filter-reducer';

describe('filter search type component', () => {
    const baseProps = {
        onChooseTypeSearch: jest.fn(),
        dataFilter: new CurrentContext(),
        type: 'beginWith',
        symbol: '*',
        filterId: 'test'
    };

    it('is shallow rendered without crashing', () => {
        const filterSearchType = <FilterSearchType {...baseProps} />;

        const filterSearchTypeComponent = shallow(filterSearchType);
        expect(filterSearchTypeComponent).toMatchSnapshot();
    });

    it('should return props symbol', () => {
        const filterSearchType = <FilterSearchType {...baseProps} />;

        const filterSearchTypeComponent = shallow(filterSearchType);
        expect(filterSearchTypeComponent.find('.filter-container__sorts-example').text()).toEqual(baseProps.symbol);
    });

    it('should hadle onChooseTypeSearch ', () => {
        const filterSearchType = <FilterSearchType {...baseProps} />;

        const filterSearchTypeComponent = shallow(filterSearchType);
        filterSearchTypeComponent.find('.filter-container__sorts-example').simulate('click');

        expect(baseProps.onChooseTypeSearch).toBeCalledWith(baseProps.filterId, baseProps.type);
    });
});
