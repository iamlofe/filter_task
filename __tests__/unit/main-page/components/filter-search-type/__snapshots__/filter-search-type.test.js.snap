// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter search type component is shallow rendered without crashing 1`] = `
<div
  className="filter-container__sorts-example"
  onClick={[Function]}
>
  *
</div>
`;
